test/controllers/service_controller_test.exs

```diff
@@ -0,0 +1,93 @@
+defmodule Appointments.ServiceControllerTest do
+  use Appointments.ConnCase
+
+  alias Appointments.Service
+  import OpenmaizeJWT.Create
+
+  @invalid_attrs %{name: "testing"}
+
+  setup do
+    company = insert(:company)
+    valid_attrs = %{name: "Haircut", duration: 45, company_id: company.id}
+
+    {:ok, token} = %{
+      id: 1,
+      email: "paolo@gmail.com",
+      role: "full",
+      first_name: "p",
+      company_id: company.id,
+      company_name: company.name
+    } |> generate_token({0, 86400})
+
+    conn = conn() |> put_req_cookie("access_token", token)
+
+    {:ok, conn: conn, company: company, valid_attrs: valid_attrs}
+  end
+
+  test "lists all entries on index", %{conn: conn} do
+    conn = get conn, service_path(conn, :index)
+    assert html_response(conn, 200) =~ "Listing services"
+  end
+
+  test "renders form for new resources", %{conn: conn} do
+    conn = get conn, service_path(conn, :new)
+    assert html_response(conn, 200) =~ "New service"
+  end
+
+  test "creates resource and redirects when data is valid",
+    %{conn: conn, valid_attrs: valid_attrs} do
+    conn = post conn, service_path(conn, :create), service: valid_attrs
+    assert redirected_to(conn) == service_path(conn, :index)
+    assert Repo.get_by(Service, valid_attrs)
+  end
+
+  test "does not create resource and renders errors when data is invalid",
+    %{conn: conn} do
+    conn = post conn, service_path(conn, :create), service: @invalid_attrs
+    assert html_response(conn, 200) =~ "New service"
+  end
+
+  test "shows chosen resource", %{conn: conn, valid_attrs: valid_attrs} do
+    service = Repo.insert! Service.changeset(%Service{}, valid_attrs)
```

Comment:
have to do it in the next PR which is branched from this one - have already refactored some of this stuff there